### ISSUE TITLE
Is depthwise seprable convolution correct?

### ISSUE THREAD

**Issue by Realdr4g0n** [xh-liu/CC-FPSE#2]
Hello xh-liu

I have some question about your paper and code

When I read Xception(https://arxiv.org/abs/1610.02357) and find code about this
code of depthwise is have more parameter(group) like follow code

        self.depthwise = nn.Conv2d(nin, nin, kernel_size=kernel_size, padding=padding, **groups=nin**, bias=bias)
        self.pointwise = nn.Conv2d(nin, nout, kernel_size=1, bias=bias)

Can you explain about why this code and your code(In the depthwise conv part) is diffrent?

**Comment by xh-liu** [comment]
Hello,

The code you provided is group convolution. Our implementation is an extreme case of group convolution where the number of groups equals the number of channels. Our implementation is equivalent to figure 4 in the Xception paper.

**Comment by Realdr4g0n** [comment]
Thanks to your reply

In your paper I've mistaken you for depthwise separable conv but it is not 
you propose a conditional depthwise convolution similar to group conv.

am I right?

**Comment by xh-liu** [comment]
In my paper we use depthwise separable conv. In your code, when group=nin it's depthwise separable conv (same as what is used in my paper); when group < nin, it is group conv.

**Comment by Realdr4g0n** [comment]
Thanks to your explain

I understand what I want to know
